src/image.rs

```diff
@@ -10,41 +11,62 @@ pub struct TileImage<'c> {
 }
 
 impl<'c> TileImage<'c> {
-    pub fn new(config: &'c Config) -> Self {
-        let mut reader = match ImageReader::open(config.filename) {
-            Ok(reader) => reader,
-            Err(e) => panic!("Problem opening the image: {:?}", e),
-        };
-        // Default memory limit of 512MB is too small for level 6+ PNGs
-        reader.no_limits();
+    pub fn new(config: &'c Config, provided_image: Option<&'c DynamicImage>) -> Self {
```

Comment:
I'd have another static function for creating one, e.g. TileImage::from_image(&DynamicImage).